src/components/ComposerQuill/utils.js

```diff
@@ -1,38 +1,44 @@
+const uuidRegex = /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/;
+const mentionRegex = /@{(.+?)_(groupMention|person)_(all|[\w-]{36})}/g;
+
 export function getFirstName(name) {
   const index = name.indexOf(' ');
 
   return index > 0 ? name.substring(0, index) : name;
 }
 
 // converts a string of text into operation deltas
-export function buildContents(text) {
-  const split = text.split(/(<spark-mention [a-zA-Z0-9-='"\s]+>.+?<\/spark-mention>)/);
+// used for drafts, where the mention object is our own placeholder string
+export function buildContents(text, mentions) {
+  const split = text.split(/(@{.+?_(?:groupMention|person)_(?:all|[\w-]{36})})/);
 
   const contents = split.map((line) => {
-    // convert spark-mention into a mention delta
-    if (line.indexOf('<spark-mention ') === 0) {
-      // converts the string to a html element so we can grab the data
-      const object = new DOMParser().parseFromString(line, 'text/xml');
-      // DOMParser returns a document but we just need the first element
-      const mention = object.firstChild;
-      const objectType = mention.getAttribute('data-object-type');
-      let objectId;
-
-      if (objectType === 'person') {
-        objectId = mention.getAttribute('data-object-id');
+    const matches = mentionRegex.exec(line);
+
+    // convert our placeholder mention into a mention delta
+    if (matches && matches.length === 4) {
+      const name = matches[1];
+      const type = matches[2];
+      const id = matches[3];
+
+      if (type === 'groupMention' || type === 'person') {
+        if (id === 'all' || uuidRegex.test(id)) {
+          // if the mention list wasn't provided then go ahead and insert
+          if (!mentions || mentions.some((mention) => mention.id === id)) {
+            return {
```

Comment:
can you comment the format expected and why

Reply:
I think that makes more sense to put into the docs rather than a code comment. I can definitely make a note to add this as i'm writing it